sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -527,12 +528,24 @@ case class WholeStageCodegenRDD(@transient sc: SparkContext, var source: CodeAnd
       } catch {
         case e: Throwable =>
           if (WholeStageCodegenExec.dumpGenCodeForException) {
-            logError(s"\n${CodeFormatter.format(source)}")
+            logFormattedError(e, s"\n${CodeFormatter.format(source)}")
           }
           throw e
       }
+    }
+  }
 
+  def logFormattedError(e: Throwable, source: String): Unit = {
+    var cause = e
+    while (cause ne null) {
+      // Don't log the code when the exception is out of memory
+      if (cause.isInstanceOf[SQLException] &&
+        cause.asInstanceOf[SQLException].getSQLState == "XCL54.T") {
```

Comment:
Can also add a check for isInstanceOf[LowMemoryException]. For example it can come from column table scan which uses region iterator and not through SQL layer. Pattern matching will be cleaner.